OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -80,6 +80,10 @@ protected virtual bool CanAttack(Actor self, Target target)
 			if (!HasAnyValidWeapons(target))
 				return false;
 
+			var layer = self.Location.Layer;
+			if (layer == CustomMovementLayerType.Tunnel || layer == CustomMovementLayerType.Subterranean)
```

Comment:
Please use `mobile.CanInteractWithGroundLayer`, which is designed for cases like this.